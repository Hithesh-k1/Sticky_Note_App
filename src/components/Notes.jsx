import React from "react";
import Draggable from "react-draggable";
import "../note.css";
export default function Notes(props) {

  return (
    <div className="container">
      {
        // props.notes.length > 0 ? (
        props.notes.map((note,i) => (
          <Draggable>
            <div>
              <div className="note" key={i}>
                <i
                  onClick={() => props.deleteNote(note.id)}
                  className="fas fa-times"
                  style={{ float: "right" }}
                ></i>
                <h2>{note.title}</h2>
                <p style={{ fontSize: "25px" }}>{note.notes}</p>

                <i
                  onClick={() => {
                    props.editRow(note);
                  }}
                  className="far fa-edit"
                  style={{ height: "25px" }}
                />
              </div>
            </div>
          </Draggable>
        ))
      }

      {/* // )
      //  : (
      //   <span colSpan={3}>No notes</span>
      )} */}
    </div>
  );
}
