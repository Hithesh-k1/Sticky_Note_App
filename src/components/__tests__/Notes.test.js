import React from "react";
import { shallow } from "enzyme";
import Notes from "../Notes";

const setUp = (props = {}) => {
  const wrappedComponent = shallow(<Notes {...props} />);
  return wrappedComponent;
};

describe("App component", () => {
  let wrappedComponent;
  let props;

  beforeEach(() => {
    const usersData = [
      { id: 1, title: "Todo 1", notes: "build CI/CD pipeline using jenkins" },
      { id: 2, title: "Todo 2", notes: "Integrate Jenkins with Sonarqube " },
      { id: 3, title: "Todo 3", notes: "Deploy to AWS S3" },
    ];
  
    props = {
      editRow: jest.fn(),
      deleteNote:jest.fn(),
      notes:usersData
    };

    wrappedComponent = setUp(props);
  });
  it("should match snapshot of Notes Component", () => {
    expect(wrappedComponent).toMatchSnapshot();
  });

  it('Test click event', () => {
    const mockCallBack = jest.fn();

    // const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>));
    const aa= wrappedComponent.find('i').at(0)
    console.log(aa.debug())
    // wrappedComponent.find('i').at(0).props().onClick() ;
    // expect(mockCallBack.mock.calls.length).toEqual(1);
  });

});
