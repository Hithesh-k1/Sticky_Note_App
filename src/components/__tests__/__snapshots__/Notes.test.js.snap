// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot of Notes Component 1`] = `
<div
  className="container"
>
  <Draggable
    allowAnyClick={false}
    axis="both"
    bounds={false}
    cancel={null}
    defaultClassName="react-draggable"
    defaultClassNameDragged="react-draggable-dragged"
    defaultClassNameDragging="react-draggable-dragging"
    defaultPosition={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    disabled={false}
    enableUserSelectHack={true}
    grid={null}
    handle={null}
    offsetParent={null}
    onDrag={[Function]}
    onMouseDown={[Function]}
    onStart={[Function]}
    onStop={[Function]}
    position={null}
    transform={null}
  >
    <div>
      <div
        className="note"
        key="0"
      >
        <i
          className="fas fa-times"
          onClick={[Function]}
          style={
            Object {
              "float": "right",
            }
          }
        />
        <h2>
          Todo 1
        </h2>
        <p
          style={
            Object {
              "fontSize": "25px",
            }
          }
        >
          build CI/CD pipeline using jenkins
        </p>
        <i
          className="far fa-edit"
          onClick={[Function]}
          style={
            Object {
              "height": "25px",
            }
          }
        />
      </div>
    </div>
  </Draggable>
  <Draggable
    allowAnyClick={false}
    axis="both"
    bounds={false}
    cancel={null}
    defaultClassName="react-draggable"
    defaultClassNameDragged="react-draggable-dragged"
    defaultClassNameDragging="react-draggable-dragging"
    defaultPosition={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    disabled={false}
    enableUserSelectHack={true}
    grid={null}
    handle={null}
    offsetParent={null}
    onDrag={[Function]}
    onMouseDown={[Function]}
    onStart={[Function]}
    onStop={[Function]}
    position={null}
    transform={null}
  >
    <div>
      <div
        className="note"
        key="1"
      >
        <i
          className="fas fa-times"
          onClick={[Function]}
          style={
            Object {
              "float": "right",
            }
          }
        />
        <h2>
          Todo 2
        </h2>
        <p
          style={
            Object {
              "fontSize": "25px",
            }
          }
        >
          Integrate Jenkins with Sonarqube 
        </p>
        <i
          className="far fa-edit"
          onClick={[Function]}
          style={
            Object {
              "height": "25px",
            }
          }
        />
      </div>
    </div>
  </Draggable>
  <Draggable
    allowAnyClick={false}
    axis="both"
    bounds={false}
    cancel={null}
    defaultClassName="react-draggable"
    defaultClassNameDragged="react-draggable-dragged"
    defaultClassNameDragging="react-draggable-dragging"
    defaultPosition={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    disabled={false}
    enableUserSelectHack={true}
    grid={null}
    handle={null}
    offsetParent={null}
    onDrag={[Function]}
    onMouseDown={[Function]}
    onStart={[Function]}
    onStop={[Function]}
    position={null}
    transform={null}
  >
    <div>
      <div
        className="note"
        key="2"
      >
        <i
          className="fas fa-times"
          onClick={[Function]}
          style={
            Object {
              "float": "right",
            }
          }
        />
        <h2>
          Todo 3
        </h2>
        <p
          style={
            Object {
              "fontSize": "25px",
            }
          }
        >
          Deploy to AWS S3
        </p>
        <i
          className="far fa-edit"
          onClick={[Function]}
          style={
            Object {
              "height": "25px",
            }
          }
        />
      </div>
    </div>
  </Draggable>
</div>
`;
